build/patches/Bromite-AdBlockUpdaterService.patch

```diff
@@ -364,18 +446,73 @@ new file mode 100644
 +            return true;
 +        });
 +
++        currentIndexVersion = (TextMessagePreference) findPreference(PREF_ADBLOCK_INDEX_VERSION);
++        currentIndexVersion.setTitle(R.string.options_adblock_current_index_title);
++
 +        mAdBlockEdit = findPreference(PREF_ADBLOCK_EDIT);
-+        updateCurrentAdBlockUrl();
++        updateCurrentAdBlockSettings();
++
++        ButtonPreference startUpdateButton =
++            (ButtonPreference) findPreference(PREF_ADBLOCK_CHECK_NOW);
++        startUpdateButton.setOnPreferenceClickListener(new OnPreferenceClickListener() {
++            @Override
++            public boolean onPreferenceClick(Preference preference) {
++                CachedFeatureFlags.AdBlockStartCheck(AdBlockPreferences.this);
++                return true;
++            }
++        });
 +    }
 +
-+    private void updateCurrentAdBlockUrl() {
++    private void updateCurrentAdBlockSettings() {
 +        mAdBlockEdit.setSummary(CachedFeatureFlags.getAdBlockFiltersURL());
++
++        DateFormat df = DateFormat.getDateTimeInstance(DateFormat.MEDIUM, DateFormat.MEDIUM);
++        currentIndexVersion.setSummary(
++            CachedFeatureFlags.getAdBlockMostRecentIndexedVersion() +
++            "\n" + getString(R.string.adblock_last_checked_text) +
++            " "  + df.format(CachedFeatureFlags.getAdBlockLastUpdateUs()));
++    }
++
++    public static String GetAdBlockMessage(int event, int error) {
++        Context applicationContext = ContextUtils.getApplicationContext();
++        // see in components/component_updater/download_filters_task.h
++        // EVENT:
```

Comment:
Wouldn't it be better to specify these as constants instead?

Reply:
yes, I change it, I've already seen how it's done.
for CachedFeatureFlags.java I need to create a new patch specific file

Reply:
fixed in https://github.com/bromite/bromite/pull/905/commits/95ab44c7afab8259a0ba50108fad5e846d609eee